app/src/main/java/cs465/illinois/edu/dogonthequad/CreateMeetupActivity.java

```diff
@@ -0,0 +1,69 @@
+package cs465.illinois.edu.dogonthequad;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+
+import com.google.gson.Gson;
+
+import java.lang.reflect.Type;
+
+import static cs465.illinois.edu.dogonthequad.MapActivity.MEETUP_KEY;
+
+public class CreateMeetupActivity extends Activity implements View.OnClickListener{
+
+    public Meetup mMeetup;
+    public Class mNextActivity;
+
+    /**
+     * Sets up the CreateMeetupActivity
+     * Handles the next button behavior and loading in the recieved Meetup from the intent.
+     * @param resourceId opens the given layout resource file
+     * @param nextActivity keeps track of the next activity in the create meetup flow
+     */
+    protected void initialize(int resourceId, Class nextActivity) {
+        setContentView(resourceId);
+
+        //all CreateMeetupActivities begin by receiving an intent from the previous activity in the flow
+        //the intent contains a meetup stored as JSON
+        Intent i = getIntent();
+        String json = i.getStringExtra(MEETUP_KEY);
+        Log.d(MEETUP_KEY, "In CreateMeetup, received meetup: " + json);
+        mMeetup = new Gson().fromJson(json, Meetup.class);
+
+        mNextActivity = nextActivity;
+
+        try {
```

Comment:
I'd say the try block isn't necessary here, since the error it's designed to catch is missing code, which should be fixed before the final product.

Reply:
It protects the CreateMeetupReview from failing, because it does not have a next button.